source/solvers/navier_stokes_vof_assemblers.cc

```diff
@@ -29,34 +29,45 @@ GLSNavierStokesVOFAssemblerCore<dim>::assemble_matrix(
   const double dt  = time_steps_vector[0];
   const double sdt = 1. / dt;
 
-  // Phase values and limiters
   std::vector<double> &phase_values = scratch_data.phase_values;
-  // std::vector<double> &phase_values_m1 =
-  // scratch_data.previous_phase_values[0];
-  // std::vector<Tensor<1, dim>> &phase_gradient_values =
-  // scratch_data.phase_gradient_values;
 
   // Limit force application : not applied if cell density is below
   // density_ratio of the maximum density (e.g. when one of the fluids is air)
-  const double density_ratio      = 2;
-  double       phase_force_cutoff = 0;
+  const double density_ratio = 2;
+  double       phase_cutoff  = 0;
 
   Assert(
     scratch_data.properties_manager.density_is_constant(),
     RequiresConstantDensity(
       "GLSVansAssemblerDiFelice<dim>::calculate_particle_fluid_interactions"));
 
+  // Phase cut-off for force (i.e. gravity) application and continuity condition
   if (scratch_data.density_0[0] < scratch_data.density_1[0] &&
       scratch_data.density_0[0] * density_ratio < scratch_data.density_1[0])
     {
-      // gravity not will be applied for phase < phase_force_cutoff
-      phase_force_cutoff = 1e-6;
+      phase_cutoff = 1e-6;
     }
   if (scratch_data.density_1[0] < scratch_data.density_0[0] &&
       scratch_data.density_1[0] * density_ratio < scratch_data.density_0[0])
     {
-      // gravity not will be applied for phase > phase_force_cutoff
-      phase_force_cutoff = 1 - 1e-6;
+      phase_cutoff = 1 - 1e-6;
+    }
+
+
+  // Determine whether continuity condition is solved in this cell
+  auto max_phase_cell =
+    std::max_element(std::begin(phase_values), std::end(phase_values));
+  bool solve_continuity(true);
+
+  if (vof_parameters.conservation.skip_mass_conservation_fluid_0)
+    {
+      if (*max_phase_cell < phase_cutoff)
+        solve_continuity = false;
```

Comment:
I couldn't find anything in the PR to explain why we need this? I'm a little confused here.